### PR TITLE
SSL on 8151

### PR DESCRIPTION
**Pull request by smcclellan**
This modifies the defaults for razor-client to use SSL on port 8151
and non-SSL on 8150. It also uses the puppet master's issued CA
if available.

For https://tickets.puppetlabs.com/browse/RAZOR-487

### CODE REVIEW

**Comment by holguinj** [comment]
:+1:

File: lib/razor/cli/navigate.rb [hunk 1]
```diff
@@ -13,6 +13,7 @@ def initialize(parse, segments)
       @doc = entrypoint
       @username, @password = parse.api_url.userinfo.to_s.split(':')
       @doc_resource = create_resource parse.api_url, {:accept => :json}
+      @ssl_ca_file = '/etc/puppetlabs/puppet/ssl/certs/ca.pem' if File.exists?('/etc/puppetlabs/puppet/ssl/certs/ca.pem')
```
Comment (smcclellan) [inline lib/razor/cli/navigate.rb]: Check to see if there's a way to make this configurable.

**Comment by lutter** [comment]
Looks good. I haven't tried it out, but I assume that if there is no CA file, the request will succeed, and the client will just accept whatever cert the server presents, correct ?

What should users do if they want to use a CA file that they have manually put somewhere, e.g. on their laptop ?

**Comment by smcclellan** [comment]
@lutter: If there is no CA file and the request is over HTTPS, the certificate will be validated (and found insecure) unless they add `-k` to the `razor` command.

Right now, that path is hard-coded as /etc/puppetlabs/puppet/ssl/certs/ca.pem, but it might also work to use a more standard path near the config file, then use a symlink to "move" the PE CA to that path.

**Comment by smcclellan** [comment]
Added "RAZOR_CA_FILE" environment variable support and pulled the /etc/puppetlabs/puppet/ssl/certs/ca.pem reference out.

**Pull request merged**